src/mbedtls/pk.pyx

```diff
@@ -1036,3 +1037,76 @@ cdef class ECDHClient(ECDHBase):
         cdef const unsigned char* end = &buffer[-1] + 1
         check_error(_pk.mbedtls_ecdh_read_params(
             &self._ctx, &first, end))
+
+
+cdef class ECDHNaive(ECDHBase):
+
+    """Naive ECDH key exchange.
+
+    Args:
+        (Curve, optional): b'curve25519' or b'curve448'.
+
+    """
+    def __init__(self, curve=None):
+        super().__init__()
+        self.curve = curve or b'curve25519'
```

Comment:
I would prefer:
```python
def __init__(self, curve=None):
    if curve is None:
        curve = b"curve25519"
    super().__init__(curve)
    ...
```
as that would be more consistent with the code in ```ECDHBase```.

Reply:
A little mistake here. It is purposely to override `__init__` from super class, `super().__init__()` should not be called here. I'll remove it later.
`ECDHBase.__init__` invokes `curve_name_to_grp_id` to get the `mbedtls_ecp_group_id` used by `mbedtls_ecp_group_load`. `curve_name_to_grp_id` will search curve name againt `mbedtls_ecp_curve_list`. This is equivalent to do a runtime curve availability check which is very good. But according to ARMmbed/mbedtls#464, CURVE25519 and CURVE448 were still missing and could not be used to do "normal" key exchange.

Reply:
Very well.